### ISSUE TITLE
Add contributor guide

### ISSUE THREAD

**Issue by janosh** [materialsproject/crystaltoolkit#269]
Since less people are familiar with `poetry` than `pip`, it would help if `contributing.md` contained steps on how to setup an editable local install of `crystal_toolkit`.

**Comment by mkhorton** [comment]
True, I don't have an issue for this, but the choice to use `poetry` has not been without issue, but I think we'll stick with it for now.

The two basic strategies are either to:

* Use `poetry` as intended, with `poetry install` and then `poetry run python your_app.py` etc.
* Use `poetry` itself or a tool like `dephell` to export a requirements.txt and/or a setup.py, for people who are more comfortable with that (we briefly tried having these alternatives checked into the repo, but it caused more confusion)

**Comment by janosh** [comment]
Small update: `poetry` was dropped in favor of `pip-tools` in https://github.com/materialsproject/crystaltoolkit/pull/292 but would still be good to have a contributing guide for newcomers.
